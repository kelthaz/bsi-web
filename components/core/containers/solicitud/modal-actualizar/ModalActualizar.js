import { useRouter } from 'next/router';
import { useDispatch, useSelector } from 'react-redux';
import { useEffect, useState } from 'react';
import Modal from '../../../../shared/modal/Modal';
import { resetChangePage, onHandleSubmit } from '../../../../../redux/actions/formulario';

const ModalActualizar = () => {
  const { push } = useRouter();
  const dispatch = useDispatch();
  const { routePage, showModal, isValid } = useSelector((state) => state.formulario);
  const [openModal, setOpenModal] = useState(false);

  const handleModalFirstOption = () => {
    setOpenModal(false);
    if (isValid) {
      dispatch(onHandleSubmit(true));
    } else {
      dispatch(resetChangePage());
    }
  };

  const handleModalSecondOption = async () => {
    setOpenModal(false);
    dispatch(resetChangePage());
    await push(routePage);
  };

  useEffect(() => {
    if (showModal) {
      setOpenModal(true);
    }
  }, [showModal]);

  return (
    <Modal openModal={openModal} setOpenModal={setOpenModal} onClose={() => dispatch(resetChangePage())}>
      <div className="modal-alerta-container">
        <h4 className="color-blue-storm">
          {isValid
            ? 'Se han detectado cambios en formulario...'
            : 'Se han detectado cambios con errores en el formulario...'}
        </h4>
        <p className="dark-gray body2">
          {isValid
            ? 'Desea guardar los cambios o descartarlos.'
            : 'Desea permanecer y corregir los errores o desea descartar los cambios.'}
        </p>
        <div className="flex-row-center-config">
          <button type="button" className="btn-medium-secondary mr-2" onClick={handleModalFirstOption}>
            {isValid ? 'Sí, guardar' : 'Sí, corregir'}
          </button>

          <button type="button" className="btn-medium ml-2" onClick={handleModalSecondOption}>
            No, descartar
          </button>
        </div>
      </div>
    </Modal>
  );
};

export default ModalActualizar;
