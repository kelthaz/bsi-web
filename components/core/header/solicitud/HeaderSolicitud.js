import React from 'react';

const HeaderSolicitud = () => {
  return <div></div>;
};

export default HeaderSolicitud;
