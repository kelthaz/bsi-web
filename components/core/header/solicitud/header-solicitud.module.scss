@import '../../../../styles/scss/modules/var';
@import '../../../../styles/scss/modules/mixins';

.container-header {
  align-items: center;
  border-bottom: 1px solid $blue-shadow;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  padding: 16px 32px;

  @include media('sm') {
    padding: 14px 16px;
  }
}
