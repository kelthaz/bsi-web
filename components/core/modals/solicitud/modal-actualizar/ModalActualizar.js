import { useRouter } from 'next/router';
import { useSelector } from 'react-redux';
import Modal from '../../../../shared/modal/Modal';

const ModalActualizar = ({ openModal, setOpenModal, formulario }) => {
  const { push } = useRouter();
  const { routePage } = useSelector((state) => state.formulario);

  const handleModalFirstOption = () => {
    if (formulario.isValid) {
      formulario.handleSubmit();
      push('/solicitud/[tab]/[step]', routePage);
    }
    setOpenModal(false);
  };

  const handleModalSecondOption = () => {
    push('/solicitud/[tab]/[step]', routePage);
    setOpenModal(false);
  };

  return (
    <Modal openModal={openModal} setOpenModal={setOpenModal}>
      <div className="modal-alerta-container">
        <h4 className="color-blue-storm">
          {formulario.isValid
            ? 'Se han detectado cambios en formulario...'
            : 'Se han detectado cambios con errores en el formulario...'}
        </h4>
        <p className="dark-gray body2">
          {formulario.isValid
            ? 'Desea guardar los cambios o descartarlos.'
            : 'Desea permanecer y corregir los errores o desea descartar los cambios.'}
        </p>
        <div className="flex-row-center-config">
          <button type="button" className="btn-medium-secondary mr-2" onClick={handleModalFirstOption}>
            {formulario.isValid ? 'Sí, guardar' : 'Sí, corregir'}
          </button>

          <button type="button" className="btn-medium ml-2" onClick={handleModalSecondOption}>
            No, descartar
          </button>
        </div>
      </div>
    </Modal>
  );
};

export default ModalActualizar;
