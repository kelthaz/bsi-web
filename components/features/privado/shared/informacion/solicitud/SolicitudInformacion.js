import React from 'react';
import PropTypes from 'prop-types';
import { MORAL } from '../../../../../../constants/persona';

const SolicitudInformacion = ({ informacionSolicitud, tipoPersona }) => {
  const labels = {
    usoCredito: 'Destino del crédito',
    descripcionCredito: 'Razón social',
    domicilioFiscal: 'Domicilio fiscal',
    domicilioComercial: 'Domicilio comercial',
    nombreCompletoRecibe: 'Nombre de quien podría recibir token',
    celularRecibe: 'Teléfono de quien podría recibir token',
    telefonoEmpresa: 'Teléfono del negocio',
    curp: tipoPersona === MORAL ? 'CURP' : 'CURP del representante legal',
    estadoCivil: 'Estado civil del solicitante',
    nombreConyuge: 'Nombre de la Pareja',
    numeroEmpleados: 'Número de empleados',

    montoSolicitado: 'Crédito solicitado',
    montoAprobado: 'Crédito aprobado',

    tasaOrdinaria: 'Tasa ordinaria',
    tasaMoratoria: 'Tasa moratoria',
    comisionApertura: 'Comisión por apertura',
    plazo: 'Plazo del crédito',
    periodicidad: 'Esquema de pago',
    pago: 'Pagos',
    cat: 'CAT',
    tieneCuentaBancoppel: 'Tiene cuenta BanCoppel',
    clabe: 'Número de cuenta BanCoppel',
  };

  if (tipoPersona === MORAL) {
    delete labels.estadoCivil;
    delete labels.nombreConyuge;
  }

  const itemsPerfil = Object.keys(labels);

  return (
    <>
      {itemsPerfil.map((item) => (
        <div key={item} className="row pb-3">
          <div className="col-6">
            <div className="body2">{labels[item]}</div>
          </div>
          <div className="col-6">
            <div className="color-blue-storm body2">{informacionSolicitud[item]}</div>
          </div>
        </div>
      ))}
    </>
  );
};

SolicitudInformacion.propTypes = {
  informacionSolicitud: PropTypes.object.isRequired,
  tipoPersona: PropTypes.string.isRequired,
};

export default SolicitudInformacion;
