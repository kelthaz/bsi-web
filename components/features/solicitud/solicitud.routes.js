import cargaDocumentosRoutes from './carga-documentos/carga-documentos.routes';
import datosEmpresaRoutes from './datos-empresa/datos-empresa.routes';
import datosPersonalesRoutes from './datos-personales/datos-personales.routes';
import documentacionRoutes from './documentacion/documentacion.routes';
import ofertaRoutes from './oferta/oferta.routes';

const solicitudRoutes = [
  ...datosPersonalesRoutes,
  // ...datosEmpresaRoutes,
  // ...documentacionRoutes,
  // ...cargaDocumentosRoutes,
  // ...ofertaRoutes
].map((route) => ({ ...route, feature: 'solicitud' }));

export default solicitudRoutes;
