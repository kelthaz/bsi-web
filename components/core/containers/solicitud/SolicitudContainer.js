import { useRouter } from 'next/router';
import PropTypes from 'prop-types';
import { useEffect } from 'react';
import { useSelector } from 'react-redux';
import { DATO_PERSONA } from '../../../../constants/formularios';
import { SIMULADOR_ROUTE } from '../../../../constants/routes/publico/publico';
import solicitudRoutes from '../../../features/solicitud/solicitud.routes';
import Step from '../../../shared/step/Step';
import TabInformativo from '../../../shared/tab-informativo/TabInformativo';
import SvgPatronesSolicitud from '../../../svgs/SvgPatronesSolicitud';
import HeaderSolicitud from '../../header/solicitud/HeaderSolicitud';
import ModalActualizar from '../../modals/solicitud/modal-actualizar/ModalActualizar';

const SolicitudContainer = ({ pageComponent, servicesData }) => {
  const { component, data } = pageComponent;
  const [componentPFAE, componentPM] = component;
  const { formulario, paso, step, tab, tipoPersona } = data;
  const { sincronizado, currentStep, tipoPersonaRedux } = useSelector((state) => state.solicitud);
  const { replace } = useRouter();

  let Component = <></>;

  if (tipoPersona === '' && componentPM) {
    Component = tipoPersonaRedux === 'MORAL' ? componentPM : componentPFAE;
  } else {
    Component = componentPFAE;
  }

  const componentsFormulario = solicitudRoutes
    .filter(({ data: dataComponent }) => dataComponent.formulario === formulario)
    .map(({ data: dataComponent, route }) => ({
      step: dataComponent.step,
      paso: dataComponent.paso,
      tipoPersona: dataComponent.tipoPersona,
      action: route,
    }));

  const steps = componentsFormulario.filter(
    (dataComponent) =>
      Number.isInteger(dataComponent.step) &&
      (dataComponent.tipoPersona === '' || dataComponent.tipoPersona === tipoPersonaRedux)
  );

  const tabs = [
    { path: 'datos-personales', label: 'Datos personales' },
    { path: 'datos-empresa', label: 'Datos de tu empresa' },
    { path: 'oferta', label: 'Oferta' },
    { path: 'documentacion', label: 'DocumentaciÃ³n' },
  ];

  useEffect(() => {
    if (!sincronizado && formulario === DATO_PERSONA) {
      replace(SIMULADOR_ROUTE);
    }
  }, []);

  return (
    <>
      <HeaderSolicitud />
      <ModalActualizar />

      {!!tab && (
        <TabInformativo
          tabs={tabs}
          currentTab={tab}
          currentStep={paso}
          valipStep={currentStep.paso}
          steps={componentsFormulario}
        />
      )}

      {!!step && <Step currentStep={step} valipStep={currentStep.valipStep} steps={steps} />}
      <Component {...servicesData} validate={currentStep.paso === paso} />
      <SvgPatronesSolicitud className="fixed-left-bottom" />
    </>
  );
};

SolicitudContainer.propTypes = {
  pageComponent: PropTypes.any.isRequired,
  servicesData: PropTypes.any.isRequired,
};

export default SolicitudContainer;
