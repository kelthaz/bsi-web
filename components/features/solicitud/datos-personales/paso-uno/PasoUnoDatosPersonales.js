import { useDispatch, useSelector } from 'react-redux';
import { useFormik } from 'formik';
import * as Yup from 'yup';
import PropTypes from 'prop-types';
import { nextStepDatosPersonales } from '../../../../../redux/actions/solicitud';
import TextField from '../../../../shared/text-field/TextField';
import { longitudMaxima, campoRequerido } from '../../../../../constants/errors';
import useOnChangePage from '../../../../../hooks/useOnChangePage';
import { PASO_DOS_DATOS_PERSONA_ROUTE } from '../../../../../constants/routes/solicitud/persona';

const PasoUnoDatosPersonales = ({ validate }) => {
  const { currentStep, datosPersonales } = useSelector((state) => state.solicitud);
  const dispatch = useDispatch();

  const formulario = useFormik({
    initialValues: {
      primerNombre: datosPersonales.primerNombre,
      segundoNombre: datosPersonales.segundoNombre,
      primerApellido: datosPersonales.primerApellido,
      segundoApellido: datosPersonales.segundoApellido,
    },
    validationSchema: Yup.object().shape({
      primerNombre: Yup.string().trim().max(60, longitudMaxima).required(campoRequerido),
      segundoNombre: Yup.string().max(60, longitudMaxima),
      primerApellido: Yup.string().trim().max(60, longitudMaxima).required(campoRequerido),
      segundoApellido: Yup.string().max(60, longitudMaxima),
    }),
    onSubmit: (values) => {
      dispatch(
        nextStepDatosPersonales({
          currentStep: validate
            ? { ...currentStep, paso: currentStep.paso + 1, valipStep: currentStep.valipStep + 1 }
            : { ...currentStep },
          datosPersonales: { ...datosPersonales, ...values },
        })
      );
    },
  });

  const [handleSubmit] = useOnChangePage(formulario, PASO_DOS_DATOS_PERSONA_ROUTE, validate);

  return (
    <div className="contedor-fixed-tab">
      <div className="contedor-solicitud">
        <div className="container p-0">
          <form onSubmit={handleSubmit} noValidate>
            <h2 className="color-blue-storm">Para comenzar</h2>
            <p className="color-dark-gray sub">Cuéntanos, ¿Cómo te llamas?</p>

            <div className="row no-gutters">
              <div className="col-lg-4 col-md-4 col-sm-12 col-xs-12 ">
                <p className="input color-gray">Mi nombre es</p>
              </div>
              <div className="col-lg-4 col-md-4 col-sm-12 col-xs-12 pr-lg-2 pr-md-2">
                <TextField
                  name="primerNombre"
                  format="uppercase"
                  maxlength={60}
                  type="text"
                  size="big"
                  label="Nombre"
                  {...formulario.getFieldMeta('primerNombre')}
                  {...formulario.getFieldHelpers('primerNombre')}
                />
              </div>
              <div className="col-lg-4 col-md-4 col-sm-12 col-xs-12">
                <TextField
                  name="segundoNombre"
                  format="uppercase"
                  maxlength={60}
                  type="text"
                  size="big"
                  label="Nombre"
                  optional
                  {...formulario.getFieldMeta('segundoNombre')}
                  {...formulario.getFieldHelpers('segundoNombre')}
                />
              </div>
            </div>
            <div className="row no-gutters">
              <div className="col-lg-6 col-md-6 col-sm-12 col-xs-12 pr-lg-2 pr-md-2">
                <TextField
                  name="primerApellido"
                  format="uppercase"
                  maxlength={60}
                  type="text"
                  size="big"
                  label="Apellido paterno"
                  {...formulario.getFieldMeta('primerApellido')}
                  {...formulario.getFieldHelpers('primerApellido')}
                />
              </div>
              <div className="col-lg-6 col-md-6 col-sm-12 col-xs-12">
                <TextField
                  name="segundoApellido"
                  format="uppercase"
                  maxlength={60}
                  type="text"
                  size="big"
                  label="Apellido materno"
                  {...formulario.getFieldMeta('segundoApellido')}
                  {...formulario.getFieldHelpers('segundoApellido')}
                />
              </div>
            </div>
            <div className="row no-gutters card-simple-blue-light text-md-center ">
              <p>Favor de llenar los campos con los nombres y apellidos presentados en tu Identificación oficial INE</p>
            </div>
            <div className="flex-column-center-config">
              <button
                type="submit"
                className="cicle-button-blue my-3"
                aria-label="Avanzar"
                disabled={validate && !(formulario.isValid && formulario.dirty)}
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  );
};

PasoUnoDatosPersonales.propTypes = {
  validate: PropTypes.bool.isRequired,
};

export default PasoUnoDatosPersonales;
