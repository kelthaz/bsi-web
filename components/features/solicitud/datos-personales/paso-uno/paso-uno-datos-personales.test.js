import '@testing-library/jest-dom';
import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { useRouter } from 'next/router';
import { act } from 'react-dom/test-utils';
import storeTest from '../../../../../redux/storeTest';
import PasoUnoDatosPersonales from './PasoUnoDatosPersonales';
import { PASO_DOS_DATOS_PERSONA_ROUTE } from '../../../../../constants/routes/solicitud/persona';
import ModalActualizar from '../../../../core/containers/solicitud/modal-actualizar/ModalActualizar';

describe('Pruebas en el componente PasoUnoDatosPersonales', () => {
  const store = storeTest();
  const push = jest.fn();
  const beforePopState = jest.fn();
  useRouter.mockImplementation(() => ({ push, beforePopState }));

  beforeEach(() => {
    jest.clearAllMocks();
  });

  test('Debe de mostrarse correctamente el componente', async () => {
    const wrapper = mount(
      <Provider store={store}>
        <PasoUnoDatosPersonales validate={false} />
        <ModalActualizar />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
    wrapper.unmount();
  });

  test('Debe de guardar en el reducer el nuevo paso y enviarlo al siguiente paso', async () => {
    // arrange
    const {
      solicitud: { currentStep },
    } = store.getState();
    const wrapper = mount(
      <Provider store={store}>
        <PasoUnoDatosPersonales validate />
      </Provider>
    );
    const form = wrapper.find('form');
    const primerNombre = wrapper.find('input').find({ name: 'primerNombre' });
    const primerApellido = wrapper.find('input').find({ name: 'primerApellido' });
    // act
    await act(async () => {
      primerNombre.simulate('change', {
        target: { value: 'alessa', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });
    await act(async () => {
      primerApellido.simulate('change', {
        target: { value: 'bolivar', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });
    await act(async () => {
      form.simulate('submit');
    });

    const {
      solicitud: { datosPersonales },
    } = store.getState();
    // assert
    expect(datosPersonales.primerNombre).toEqual(wrapper.find('input').find({ name: 'primerNombre' }).prop('value'));
    expect(datosPersonales.primerApellido).toEqual(
      wrapper.find('input').find({ name: 'primerApellido' }).prop('value')
    );
    expect(store.getState().solicitud.currentStep.paso).toBe(currentStep.paso + 1);
    expect(store.getState().solicitud.currentStep.valipStep).toBe(currentStep.valipStep + 1);
    expect(push).toHaveBeenCalledTimes(1);
    expect(push).toHaveBeenCalledWith(PASO_DOS_DATOS_PERSONA_ROUTE);
    wrapper.unmount();
  });

  test('Debe de mantener el paso anterior en el reducer y enviarlo al siguiente paso', async () => {
    // arrange
    const {
      solicitud: { currentStep },
    } = store.getState();

    const wrapper = mount(
      <Provider store={store}>
        <PasoUnoDatosPersonales validate={false} />
        <ModalActualizar />
      </Provider>
    );

    const form = wrapper.find('form');
    const primerNombre = wrapper.find('input').find({ name: 'primerNombre' });
    const primerApellido = wrapper.find('input').find({ name: 'primerApellido' });
    // act
    await act(async () => {
      primerNombre.simulate('change', {
        target: { value: 'santiago', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });
    await act(async () => {
      primerApellido.simulate('change', {
        target: { value: 'perez', selectionStart: 0, setSelectionRange: jest.fn() },
      });
    });
    await act(async () => {
      form.simulate('submit');
    });
    wrapper.update();
    await act(async () => {
      wrapper.find('.btn-medium-secondary').simulate('click');
    });

    const {
      solicitud: { datosPersonales },
    } = store.getState();

    // assert
    expect(datosPersonales.primerNombre).toEqual(wrapper.find('input').find({ name: 'primerNombre' }).prop('value'));
    expect(datosPersonales.primerApellido).toEqual(
      wrapper.find('input').find({ name: 'primerApellido' }).prop('value')
    );
    expect(store.getState().solicitud.currentStep.paso).toBe(currentStep.paso);
    expect(store.getState().solicitud.currentStep.valipStep).toBe(currentStep.valipStep);
    expect(push).toHaveBeenCalledTimes(1);
    expect(push).toHaveBeenCalledWith(PASO_DOS_DATOS_PERSONA_ROUTE);
    wrapper.unmount();
  });
});
