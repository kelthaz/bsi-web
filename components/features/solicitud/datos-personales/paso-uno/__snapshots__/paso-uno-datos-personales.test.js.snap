// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componente PasoUnoDatosPersonales Debe de mostrarse correctamente el componente 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <PasoUnoDatosPersonales
    validate={false}
  >
    <div
      className="contedor-fixed-tab"
    >
      <div
        className="contedor-solicitud"
      >
        <div
          className="container p-0"
        >
          <form
            noValidate={true}
            onSubmit={[Function]}
          >
            <h2
              className="color-blue-storm"
            >
              Para comenzar
            </h2>
            <p
              className="color-dark-gray sub"
            >
              Cuéntanos, ¿Cómo te llamas?
            </p>
            <div
              className="row no-gutters"
            >
              <div
                className="col-lg-4 col-md-4 col-sm-12 col-xs-12 "
              >
                <p
                  className="input color-gray"
                >
                  Mi nombre es
                </p>
              </div>
              <div
                className="col-lg-4 col-md-4 col-sm-12 col-xs-12 pr-lg-2 pr-md-2"
              >
                <TextField
                  capitalize={false}
                  disabled={false}
                  error=""
                  format="uppercase"
                  initialTouched={false}
                  initialValue="JESUS"
                  inverted={false}
                  label="Nombre"
                  maxlength={60}
                  name="primerNombre"
                  optional={false}
                  paste={true}
                  readonly={false}
                  setError={[Function]}
                  setTouched={[Function]}
                  setValue={[Function]}
                  size="big"
                  touched={false}
                  type="text"
                  value="JESUS"
                >
                  <div
                    className="group"
                  >
                    <input
                      autoComplete="off"
                      className="input-big  "
                      disabled={false}
                      id="primerNombre"
                      maxLength={60}
                      name="primerNombre"
                      onBeforeInput={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      onPaste={[Function]}
                      placeholder="Nombre"
                      readOnly={false}
                      tabIndex="0"
                      type="text"
                      value="JESUS"
                    />
                    <span>
                       
                    </span>
                  </div>
                </TextField>
              </div>
              <div
                className="col-lg-4 col-md-4 col-sm-12 col-xs-12"
              >
                <TextField
                  capitalize={false}
                  disabled={false}
                  error=""
                  format="uppercase"
                  initialTouched={false}
                  initialValue=""
                  inverted={false}
                  label="Nombre"
                  maxlength={60}
                  name="segundoNombre"
                  optional={true}
                  paste={true}
                  readonly={false}
                  setError={[Function]}
                  setTouched={[Function]}
                  setValue={[Function]}
                  size="big"
                  touched={false}
                  type="text"
                  value=""
                >
                  <div
                    className="group"
                  >
                    <input
                      autoComplete="off"
                      className="input-big  "
                      disabled={false}
                      id="segundoNombre"
                      maxLength={60}
                      name="segundoNombre"
                      onBeforeInput={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      onPaste={[Function]}
                      placeholder="Nombre"
                      readOnly={false}
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <span>
                      Opcional
                       
                    </span>
                  </div>
                </TextField>
              </div>
            </div>
            <div
              className="row no-gutters"
            >
              <div
                className="col-lg-6 col-md-6 col-sm-12 col-xs-12 pr-lg-2 pr-md-2"
              >
                <TextField
                  capitalize={false}
                  disabled={false}
                  error=""
                  format="uppercase"
                  initialTouched={false}
                  initialValue="DAVID"
                  inverted={false}
                  label="Apellido paterno"
                  maxlength={60}
                  name="primerApellido"
                  optional={false}
                  paste={true}
                  readonly={false}
                  setError={[Function]}
                  setTouched={[Function]}
                  setValue={[Function]}
                  size="big"
                  touched={false}
                  type="text"
                  value="DAVID"
                >
                  <div
                    className="group"
                  >
                    <input
                      autoComplete="off"
                      className="input-big  "
                      disabled={false}
                      id="primerApellido"
                      maxLength={60}
                      name="primerApellido"
                      onBeforeInput={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      onPaste={[Function]}
                      placeholder="Apellido paterno"
                      readOnly={false}
                      tabIndex="0"
                      type="text"
                      value="DAVID"
                    />
                    <span>
                       
                    </span>
                  </div>
                </TextField>
              </div>
              <div
                className="col-lg-6 col-md-6 col-sm-12 col-xs-12"
              >
                <TextField
                  capitalize={false}
                  disabled={false}
                  error=""
                  format="uppercase"
                  initialTouched={false}
                  initialValue=""
                  inverted={false}
                  label="Apellido materno"
                  maxlength={60}
                  name="segundoApellido"
                  optional={false}
                  paste={true}
                  readonly={false}
                  setError={[Function]}
                  setTouched={[Function]}
                  setValue={[Function]}
                  size="big"
                  touched={false}
                  type="text"
                  value=""
                >
                  <div
                    className="group"
                  >
                    <input
                      autoComplete="off"
                      className="input-big  "
                      disabled={false}
                      id="segundoApellido"
                      maxLength={60}
                      name="segundoApellido"
                      onBeforeInput={[Function]}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      onPaste={[Function]}
                      placeholder="Apellido materno"
                      readOnly={false}
                      tabIndex="0"
                      type="text"
                      value=""
                    />
                    <span>
                       
                    </span>
                  </div>
                </TextField>
              </div>
            </div>
            <div
              className="row no-gutters card-simple-blue-light text-md-center "
            >
              <p>
                Favor de llenar los campos con los nombres y apellidos presentados en tu Identificación oficial INE
              </p>
            </div>
            <div
              className="flex-column-center-config"
            >
              <button
                aria-label="Avanzar"
                className="cicle-button-blue my-3"
                disabled={false}
                type="submit"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  </PasoUnoDatosPersonales>
  <ModalActualizar>
    <Modal
      closeable={true}
      onClose={[Function]}
      openModal={false}
      setOpenModal={[Function]}
    />
  </ModalActualizar>
</Provider>
`;
